Feature: Bingo Boop Party Invite

NetheriteMiner opened the pull request:
## What
Sends a clickable chat message when the user receives a boop prompting them to party the person who booped them, because  bingo users needed this apparently

## Changelog New Features
+ Added Bingo Boop for Party Invite. - NetheriteMiner

NetheriteMiner committed:
Add party on boop for bingo

lineargraph commented on src/main/java/at/hannibal2/skyhanni/features/bingo/BoopParty.kt:
```diff
@@ -0,0 +1,37 @@
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
```
You could use a `PrivateMessageChatEvent` here.

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/bingo/BoopParty.kt:
```diff
@@ -0,0 +1,37 @@
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class BoopParty {
```
should be an object with @SkyHanniModule and then should not be loaded in SkyhanniMod

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/bingo/BoopParty.kt:
```diff
@@ -0,0 +1,37 @@
+class BoopParty {
+
+    val config get() = SkyHanniMod.feature.event.bingo.boopParty
+    val patternGroup = RepoPattern.group("misc")
```
group should be more descriptive

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/bingo/BoopParty.kt:
```diff
@@ -0,0 +1,37 @@
+        }
+    }
+
+    fun isEnabled() = config && LorenzUtils.isBingoProfile
```
should check if bingo profile first

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/NetheriteMiner/bingoboop

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/SkyHanniMod.kt

CalMWolfs committed:
fix merge

NetheriteMiner committed:
Merge branch 'hannibal002:beta' into bingoboop

NetheriteMiner committed:
Fix event and enabled order according to requested changes

NetheriteMiner committed:
Merge branch 'hannibal002:beta' into bingoboop

NetheriteMiner committed:
Change pattern group to be more descriptive

This was discussed in DMs with Cal, so there shouldn't be any problems with bingo.boop

Thunderblade73 reviewed (approved):
Code look good. (I did not test)

CalMWolfs committed:
Merge branch 'beta' into bingoboop

CalMWolfs committed:
change module name

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/bingo/BingoBoopParty.kt:
```diff
@@ -0,0 +1,39 @@
+    fun onChat(event: PrivateMessageChatEvent) {
+        if (!isEnabled()) return
+
+        val msg = event.message.trimWhiteSpaceAndResets()
+        boopPattern.matchMatcher(msg) {
```
event.message is the stuff after From name: 
so this will never match.
Instead you could check if event.message is whatever it shows when its a boop and then use event.author

NetheriteMiner committed:
Merge branch 'hannibal002:beta' into bingoboop

NetheriteMiner committed:
Fix regex never matching

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/NetheriteMiner/bingoboop

hannibal002 committed:
code cleanup
